metadata_please/source_checkout.py

```diff
@@ -213,7 +227,26 @@ def from_poetry_checkout(path: Path) -> bytes:
                 optional = False
 
             if not version:
-                # e.g. git, path or url dependencies, skip for now
+                # e.g. git, path or url dependencies
+                if "path" in v:
```

Comment:
Can you split this change out to a separate PR?  This needs more thought and probably some sort of X- prefixed name, as I'm pretty sure they are not valid Requires-Dist.